elliptic-curve/src/ops.rs

```diff
@@ -25,6 +26,77 @@ pub trait Invert {
     }
 }
 
+/// Perform a batched inversion on a sequence of field elements (i.e. base field elements or scalars)
+/// at an amortized cost that should be practically as efficient as a single inversion.
+pub trait InvertBatch: Invert + Sized {
+    /// The output type of batch inversion.
+    /// Since inversion is performed in-place, the outputted value doesn't contain inverses of field elements.
+    /// Instead, it should be set to `Choice` when inversion may fail or `()` if it always succeeds.
+    type Output;
```

Comment:
Maybe just call this `Choice`? Or always use `subtle::Choice`? What is the benefit of making it generic?

I guess it could be `()` for `NonZeroScalar` but what does that actually buy you?

Reply:
That definitely is an option. It buys me a clear API and no need to pattern match in the case of NonZeroScalar for example. I wasn't sure about it, I'll change it as you suggest